File: doc/src/scripts/CombinedIsp.cpp
#include <iostream>
#include <vector>
#include <krpc.hpp>
#include <krpc/services/space_center.hpp>

using SpaceCenter = krpc::services::SpaceCenter;

int main() {
  auto conn = krpc::connect();
  SpaceCenter sc(&conn);
  auto vessel = sc.active_vessel();

  auto engines = vessel.parts().engines();

  std::vector<SpaceCenter::Engine> active_engines;
  for (auto engine : engines)
    if (engine.active() && engine.has_fuel())
      active_engines.push_back(engine);

  std::cout << "Active engines:" << std::endl;
  for (auto engine : active_engines)
    std::cout << "   " << engine.part().title() << " in stage " << engine.part().stage() << std::endl;

  double thrust = 0;
  double fuel_consumption = 0;
  for (auto engine : active_engines) {
    thrust += engine.thrust();
    fuel_consumption += engine.thrust() / engine.specific_impulse();
  }
  double isp = thrust / fuel_consumption;
  std::cout << "Combined vacuum Isp = " << isp << " seconds" << std::endl;
}

File: doc/src/scripts/CombinedIsp.lua
local krpc = require 'krpc'
local math = require 'math'
local conn = krpc.connect()
local vessel = conn.space_center.active_vessel

local active_engines = {}
for engine in vessel.parts.engines do
    if engine.active and engine.has_fuel then
       active_engines.add(engine)
    end
end

print('Active engines:')
for engine in active_engines do
    print('   ' .. engine.part.title .. ' in stage ' .. engine.part.stage)
end

thrust = 0
fuel_consumption = 0
for engine in active_engines do
    thrust = thrust + engine.thrust
    fuel_consumption = fuel_consumption + engine.thrust / engine.specific_impulse
end
isp = thrust / fuel_consumption

print('Combined vacuum Isp = ' .. isp .. ' seconds')
